kafka-commons/src/main/java/uk/gov/homeoffice/digital/sas/kafka/validators/SchemaValidator.java

```diff
@@ -0,0 +1,92 @@
+package uk.gov.homeoffice.digital.sas.kafka.validators;
+
+import static uk.gov.homeoffice.digital.sas.kafka.constants.Constants.KAFKA_SCHEMA_INVALID;
+import static uk.gov.homeoffice.digital.sas.kafka.constants.Constants.KAFKA_SCHEMA_INVALID_RESOURCE;
+import static uk.gov.homeoffice.digital.sas.kafka.constants.Constants.KAFKA_SCHEMA_INVALID_VERSION;
+import static uk.gov.homeoffice.digital.sas.kafka.constants.Constants.KAFKA_SCHEMA_VALIDATED;
+import static uk.gov.homeoffice.digital.sas.kafka.constants.Constants.SCHEMA;
+
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Pattern;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.maven.artifact.versioning.DefaultArtifactVersion;
+
+@Slf4j
+@Getter
+@Setter
+@NoArgsConstructor
+public abstract class SchemaValidator {
+
+  String resourceName;
+
+  DefaultArtifactVersion supportedVersion;
+
+  public boolean isSchemaValid(String message) {
+    JsonObject jsonMessage = JsonParser.parseString(message).getAsJsonObject();
+    String schema = jsonMessage.get(SCHEMA).getAsString();
+    List<String> splitSchema = splitMessageSchema(schema);
+    return isValidMessage(splitSchema, schema);
+  }
+
+  private List<String> splitMessageSchema(String schema) {
+    List<String> stringList = new ArrayList<>();
+    if (schema.contains(",")) {
+      stringList = Pattern.compile(", ")
```

Comment:
would it be a good idea to have the comma string as a constant used by both the validator and the kafka event message classes?

Reply:
Moved to constants file and updated SCHEMA_FORMAT used in kafka event message to use the delimeter.